### DISCUSSION TITLE
Sound doesn't change

### DISCUSSION THREAD

**Discussion by damianevans** [rerdavies/pipedal#198] [General]
Hi 
I hope someone can help me. My set up is as follows:
Raspberry Pi 4 with Bookworm
RPI official 7" Touchscreen
AudioInjector Stereo hat

I play a sound into the input rca jack on my audioinjector hat, and I can hear the sound coming out of my speakers, but none of the settings make any change to the sound and the VU meters do not change at all. I've tried switching the Pi's audio config between pulseAudio and pipewire, but it makes no difference. Can anyone suggest anything else I can try?

**Comment by rerdavies** [comment]
PiPedal uses ALSA because neither jackd, nor pipewire nor pulseaudio can be used from a daemon.  Pros: pipedal can run as an IoT appliance without a login session; cons: no jack routing.  

You must configure jackd and.or pipewire to NOT use the device that PiPedal will use. And then reboot. And then configure PiPedal  using PiPpedal's web UI. Select the Audio Device item in PiPedal's Settings page. 

For what it's worth, if you want to switch back and forth between PiPedal and a pipwire-based host, you can use the `pipedalconfig ` comand-line utility to stop the pipedald service. Stopping the pipedal service will close the ALSA device so that pipewire can use it instead.

**Reply by damianevans** [reply]
Thanks for the reply, Robin.
I'd appreciate it if anyone can suggest the optimum set of configuration settings for RPI. So for example, in raspi-config, you have the option to set pipewire or pulseaudio, but it has to be one of them. Also I've read to disable/comment out the `dtparam=audio=on` in config.txt

So are you suggesting that we should disable the Jackd daemon altogether if we only want to use pipedal?

**Comment by rerdavies** [comment]
I'm not entirely sure about current details on bookworm. I haven't worked
with pipewire/jack in a while, and when I did, it was on bookworm. So what
follows may not reflect current reality. You also need to temper anything I
say because I have never actually worked with an internal audio hat. Just
USB audio adapters. So there may be differences due to that.

You don't have to disable any of the daemons. You just have to make sure
that PulseAudio uses some device OTHER than your pi hat. And not be using
an Jack audio application at the same time you're trying to select an audio
device in PiPedal.

You should, however, disable PulseAudio. If you're trying to do low-latency
realtime audio in jack, nothing actually good is going to come from
PulseAudi (other than getting email notification should while you're
playing live at a gig) And you should disable PulseAudio if you are using
PiPedal, even if you won't get email notifications through the PA.

There's nothing in raspi-config that's going to cause problems for PiPedal.
By default, raspi-config will configure PulseAudio to use the internal 3/8"
audio jack. However, if your audio hat has configuration utilities, then
that would be the place to look. If there is a configuration utility, then
configuring PulseAudio to use your audio hat is exactly the sort of thing
it would do. And it would (presumably) also provide a way to make
PulesAudio NOT use your audio hat.

And you should make sure that you have selected an audio device in the
Pipedal UI.

In case it's not clear, once configured, Pipedal will grab the audio device
very early in the boot process, and will not let go of it again until you
shut down your Raspberry Pi, or stop the pipedald service.  You will not be
able to see the device in Jack configuration dialogs. (And PulseAudio
shouldn't be able to see the device either, I think).

And if PulseAudio is using your audio hat, it will open the audio device
when you log in,, and not let go of it until you log out. You will not be
able to see the device in the PiPedal UI.

If that doesn't suit, you can:

     pipedalconfig --stop      # stop the service (and release the audio
device so jack can use it.

    pipedalconfig -- disable # dont' start the service after a reboot.

    pipedalconfig -- enable     # do start the service after a reboot.

    pipedalconfi --start     # start the pipedald service (grab the audio
device so jack can't use it.

In case it's not clear, when you use jack and pulseaudio, you are actually
using Pipewire's emulations of jack and pulseaudio, unless you have been
doing some truly heroic reconfiguration of Raspberry Pi Os bookworm.

There is some reason to think that the old jack2 is better than pipewire.
There was, at least, when I last researched the issue, about a year a go.
Pipewire, is -- in theory -- as good as the original jack2 from a latency
and stability perspective. In actual reality, there are  quite  few field
reports of problems with low-latency realtime audio stabiity when using
pipewire instead of jack. And a significant wave of push-back against
adopting pipewire as the default audio provider in mainline distros,
primarily because of unresolved field reports of problems with "pro-audio"
applications. And because --as things stand currenty -- pipewire doesn't
seem to actually improve anything. (Long term, it will, but at present,
there's no real additional functionality provided by pipewire, other than
*maybe* being able to route PulseAudio data into the audio chain when jack
is running, Which is more of a liability than a feature). There is (I
think) good reason to think that jack2 provides better pro-audio support
than pipewire, but that may have changed in the last year., There is no
technical reason why they should be any different. They both provide a thin
skin around ALSA audio, once the device is set up and running, and the
amount of glue code required to do so is insignificant,  You *can*
configure "bookworm" to use the old jack2 services; but it does require
pretty heroic effort to do so. Not recommended.


- R







Juse

**Comment by rerdavies** [comment]
Oh, and....

     #dtparam=audio=on

Seems like a good idea. Disable the crappy 3/8" audio jack. The only
downside I can see is that it makes it much more like that either
pipewire's PulseAudio implementation,
or the original PulseAudio implementation will grab your audio-hat as the
default audio device. In default configuration, PulseAudio will grab the
3/8" audio device.
Which is kind of a good thing because it allows both pipedal and
jack/pipewire to run without interference from PulseAudio.

And if you're looking at the page where the guy gets PulseAudio to run as a
system process shared between multiple login sessions, run don't walk in
the opposite direction.
pipewire won't like that; jack2 won't like that; and pipedal won't like
that.

Just to reiterate: PulseAudio is a very sensible option if all you want out
of your Raspberry Pi is the ability to go "bing" occasionally, when an
email arrives, or play the
"Windows startup sound" when you log in, some such.

But if you plan to use serious audio applications of any kind, PulseAudio
tends to break things.

**Comment by BorisSutin** [comment]
I have an audioinjector. Open the alsamixer, select your card and turn off all the parameters that are there, perhaps you have a digital loopback enabled
